api/fasit.go

```diff
@@ -122,68 +214,232 @@ func fetchFasitResources(fasitUrl string, deploymentRequest NaisDeploymentReques
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create payload (%s)", err)
+	}
+
+	req, err := http.NewRequest("POST", fasit.FasitUrl+"/api/v2/resources/", bytes.NewBuffer(payload))
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create request: %s", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to contact Fasit: %s", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	httpReqsCounter.WithLabelValues(strconv.Itoa(resp.StatusCode), "POST").Inc()
+	if resp.StatusCode > 299 {
+		errorCounter.WithLabelValues("error_fasit").Inc()
+		return 0, fmt.Errorf("Fasit returned: %s (%s)", body, strconv.Itoa(resp.StatusCode))
+	}
+
+	type CreatedResource struct {
+		Id int
+	}
+	createdResource := new(CreatedResource)
+	if err := json.Unmarshal(body, &createdResource); err != nil {
+		errorCounter.WithLabelValues("read_body").Inc()
+		return 0, fmt.Errorf("Could not read response: %s", err)
+	}
+
+	return createdResource.Id, nil
+}
+func (fasit FasitClient) updateResource(existingResourceId int, resource ExposedResource, fasitEnvironment, application, zone, hostname string) (int, error) {
+	requestCounter.With(nil).Inc()
+
+	payload, err := json.Marshal(buildResourcePayload(resource, fasitEnvironment, zone, hostname))
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create payload (%s)", err)
+	}
+
+	req, err := http.NewRequest("PUT", fasit.FasitUrl+"/api/v2/resources/"+fmt.Sprint(existingResourceId), bytes.NewBuffer(payload))
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to create request: %s", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		errorCounter.WithLabelValues("create_request").Inc()
+		return 0, fmt.Errorf("Unable to contact Fasit: %s", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	httpReqsCounter.WithLabelValues(strconv.Itoa(resp.StatusCode), "POST").Inc()
+	if resp.StatusCode > 299 {
+		errorCounter.WithLabelValues("error_fasit").Inc()
+		return 0, fmt.Errorf("Fasit returned: %s (%s)", body, strconv.Itoa(resp.StatusCode))
+	}
+
+	type CreatedResource struct {
+		Id int
+	}
+	createdResource := new(CreatedResource)
+	if err := json.Unmarshal(body, &createdResource); err != nil {
+		errorCounter.WithLabelValues("read_body").Inc()
+		return 0, fmt.Errorf("Could not read response: %s", err)
+	}
+
+	return createdResource.Id, nil
+}
+
+func (fasit FasitClient) GetFasitEnvironment(environmentName string) error {
+	requestCounter.With(nil).Inc()
+	req, err := http.NewRequest("GET", fasit.FasitUrl+"/api/v2/environments/"+environmentName, nil)
```

Comment:
We have a helper function  doRequest(r *http.Request) ([]byte, AppError) that could be used to avoid http boilerplate code.